lib/CMakeLists.txt

```diff
@@ -141,7 +141,7 @@ set(LLVM_INCLUDE_BENCHMARKS OFF)
 set(LLVM_INCLUDE_TESTS OFF)
 set(LLVM_TOOL_REMARKS_SHLIB_BUILD OFF)
 set(LLVM_ENABLE_ZLIB OFF)
-set(LLVM_ENABLE_ZLIB:STRING OFF)
```

Comment:
was this tested on Linux? this doesn't seem correct. that we now have LLVM_ENABLE_ZLIB OFF and "OFF".

If it was tested, how was it tested?

Reply:
Ive found something concerning with this @mfelsche. So locally using the latest pre-packaged ponyc, I can't link the "work" branch of ponydoc due to zlib linking errors, but if I build from source with the exact same LLVM settings, it works, so... there seems to be something going on that i'm unsure about but that is making testing hard and is quite possibly impacting on anyone who tests this PR (including you).

Reply:
It was tested by taking a pony program and linking it to `libponyc-standalone.a` and trying to compile it successfully.

Take this one as a minimal example:

```
use "lib:ponyc-standalone"

actor Main
  new create(env: Env) =>
    env.out.print("it works")
```

Reply:
that's not sufficient to test for linker errors. if it doesn't exercise the various functions that need say, zlib, then this won't show any linker errors. try testing with the `work` branch of ponydoc. You'll need to update the Makefile to use a your ponyc and then can do `make` to try building ponydoc. If it compiles and links you should be good at least on your test machine. 

Also note, for others in the future, if you don't have zlib installed in a way that llvm can detect then you can have incorrect entries for turning off ZLIB in the LLVM setup and you won't be able to detect in your test environment.

Reply:
> So locally using the latest pre-packaged ponyc, I can't link the "work" branch of ponydoc due to zlib linking errors

I am able to reproduce this issue. I was always testing this with the ponyc that i built in the project. The ponyc installed with ponyup didn't work for me. We can (and should) investigate this in another PR.

Reply:
> Ive found something concerning with this @mfelsche. So locally using the latest pre-packaged ponyc, I can't link the "work" branch of ponydoc due to zlib linking errors, but if I build from source with the exact same LLVM settings, it works, so... there seems to be something going on that i'm unsure about but that is making testing hard and is quite possibly impacting on anyone who tests this PR (including you).

The issue that I found that folks need to be aware of for testing is that your test environment must have Zlib installed in a way that LLVM finds it otherwise, no amount of screwing up the LLVM config will result in the errors. 

I think before we merge this PR we should have a test in place as part of our CI that verifies that we have this working for supported platforms (Linux and MacOS). There's some serious ways to mess up the CI setup though that would need to be resolved.

I think we should have that CI support as you opened this PR as there is now someone other than me who would be relying on the standalone library working and not getting broken (which appears to be easy to mess up).

Reply:
on MacOS the combination of:

```
set(LLVM_ENABLE_ZLIB OFF)
set(LLVM_ENABLE_ZLIB:STRING OFF)
```

resulted in llvm still requiring libz. Only this worked for me on MacOs:

```
set(LLVM_ENABLE_ZLIB OFF)
set(LLVM_ENABLE_ZLIB "OFF")
```

Linux still needed the version without the quotes. It could also be a difference in cmake versions used on both systems...